### DISCUSSION TITLE
Modify image of a layer in PSD file

### DISCUSSION THREAD

**Discussion by Zebra447** [ImageMagick/ImageMagick#5396] [Help]
Is it possible to use PHP+ImageMagick to replace an image inside a PSD file?

**Comment by snibgo** [comment]
IM can read images from a PSD file into memory. Then it can manipulate the images in memory. Then it can write the images to a PSD file, either a new file or overwriting the old file.

But I don't use Photoshop, and can't advise on exactly what data IM reads from a PSD, or writes to it.

**Reply by fmw42** [reply]
Zip and Post and example PSD file and explain what you want changed.

**Comment by Zebra447** [comment]
I use this code to find the layer containing the design image, but how can I replace the image inside the layer?

```
// Load the PSD file
$psd = new Imagick(realpath("130.psd"));

// Export layers
foreach ($psd as $no => $layer) {

    // Get the name of this layer
    $imageProperties = $layer->getimageproperties();

    // Find layer with tshirt design image
    if ( $imageProperties['label'] == "Design" ) {
        echo "<p>Found DESIGN label</p>";
        print_r($imageProperties);
    }

}
```

**Reply by fri3ndly** [reply]
Did you ever figure this out? 👍

**Comment by snibgo** [comment]
That seems to be a question about IMagick. Sorry, but I know nothing about IMagick.
